src/pages/docs/ai-transport/features/sessions-identity/identifying-users-and-agents.mdx

```diff
@@ -0,0 +1,449 @@
+
+**Message attribution**: Agents need to know whether messages come from users or other agents. This is essential for conversation flows, where agent responses should be securely distinguished from user prompts.
+
+**Personalized behavior**: Different users may have different privileges or attributes. A premium user might get access to more capable models, while a free user gets basic functionality. Ably allows your trusted authentication server to embed this information in the client's credentials, allowing this information to be securely passed to agents.
+
+**Authorization decisions**: Some operations should only be performed for specific users. For example, human-in-the-loop (HITL) tool calls that access sensitive data might require admin privileges. Ably allows agents to verify the privilege level and role of the user resolving the tool call.
+
+## Authenticating users <a id="authenticating"/>
+
+Use [token authentication](/docs/auth/token) to securely authenticate users. Your authentication server generates a token that is signed with the secret part of your Ably API key. Clients use this token to connect to Ably, and the token signature ensures it cannot be tampered with.
+
+The following examples use [JWT authentication](/docs/auth/token#jwt) for its simplicity and standard tooling support. For other approaches, see [token authentication](/docs/auth/token).
+
+Create a server endpoint that generates signed JWTs after verifying user authentication:
+
+<Code>
+```javascript
+// Server code
+import express from "express";
+import jwt from "jsonwebtoken";
+
+const app = express();
+
+// Mock authentication middleware.
+// This should be replaced with your actual authentication logic.
+function authenticateUser(req, res, next) {
+  // Assign a mock user ID for demonstration
+  req.session = { userId: "user123" };
+  next();
+}
+
+// Return the claims payload to embed in the signed JWT.
+function getJWTClaims(userId) {
+  // Returns an empty payload, so the token
+  // inherits the capabilities of the signing key.
+  return {};
+}
+
+// Define an auth endpoint used by the client to obtain a signed JWT
+// which it can use to authenticate with the Ably service.
+app.get("/api/auth/token", authenticateUser, (req, res) => {
+  const [keyName, keySecret] = "{{API_KEY}}".split(":");
+
+  // Sign a JWT using the secret part of the Ably API key.
+  const token = jwt.sign(getJWTClaims(req.session.userId), keySecret, {
+    algorithm: "HS256",
+    keyid: keyName,
+    expiresIn: "1h",
+  });
+
+  res.type("application/jwt").send(token);
+});
+
+app.listen(3001);
+```
+</Code>
+
+The JWT is signed with the secret part of your Ably API key using HMAC SHA-256. This example does not embed any claims in the JWT payload, so by default the token inherits the capabilities of the Ably API key used to sign the token.
```

Comment:
I think it's `HMAC-SHA-256` - and should it link somewhere? eg https://datatracker.ietf.org/doc/html/rfc4868